### ISSUE TITLE
`Type instantiation is excessively deep and possibly infinite` with RPC

### ISSUE THREAD

**Issue by subhendupsingh** [honojs/hono#1360]
### What version of Hono are you using?

3.2.5

### What runtime/platform is your app running on?

Cloudflare Workers

### What steps can reproduce the bug?

When the routes are separated into different files and the endpoints are defined using the builder pattern, for a route that has 17 methods and 378 lines of code

```
const organizationRoute = organizationController

.get("/layout", async (c, next) => {
    const response = await getOrganizationLayout(c);
    return controllerResponse(response, c);
})

.get("/link", checkAccess(), async (c) => {

})

export { organizationRoute };
```

Typescript gives this compile time error.

### What is the expected behavior?

The generated types should be simplified to avoid the error.

### What do you see instead?

<img width="860" alt="image" src="https://github.com/honojs/hono/assets/5727516/4f1ab7fa-7070-4f8b-bf6c-0ad232643ea5">

Nested types generated cause the problem.

**Comment by yusukebe** [comment]
Hi @subhendupsingh !

This may be fixed with new version "v3.5.2". Try it and share the whether it is resolved.

**Comment by Tanner-Scadden** [comment]
@yusukebe This worked great on the backend. On the frontend it throws this error now. Is there a version of Typescript you'd recommend being on?
